### PR TITLE
added flexiblity and robustness to timezone tools

### PR DESCRIPTION
**Pull request by collinbrake**

### CODE REVIEW

**Comment by collinbrake** [comment]
@cbrake I tried to incorporate your suggestions. Let me know what you think.

File: system/timezone.go [hunk 1]
```diff
@@ -36,26 +70,13 @@ func SetTimezone(zone string) error {
...
-		log.Println("Error reading time zones, ", err)
-		return nil, err
-	}
+// Symbolic link for the system timezone
+const zoneLink = "/etc/localtime"
 
-	for _, fi := range fileInfo {
-		list = append(list, fi.Name())
+func setZoneInfoPath(zoneInfoDir string) (zoneInfoPath string) {
+	zoneInfoPath = "/usr/share/zoneinfo/"
```
Comment (cbrake) [inline system/timezone.go]: consider using path.Join

File: system/timezone.go [hunk 2]
```diff
@@ -17,17 +49,19 @@ func GetTimezone() (string, error) {
...
+
+	if _, err := os.Lstat(zoneLink); err == nil {
+		err := os.Remove(zoneLink)
 		if err != nil {
 			log.Println("Error removing old time zone link, ", err)
 			return err
 		}
 	}
 
-	err := os.Symlink("/usr/share/zoneinfo/US/"+zone, "/etc/localtime")
```
Comment (cbrake) [inline system/timezone.go]: use path.Join here

File: system/timezone.go [hunk 3]
```diff
@@ -3,31 +3,80 @@ package system
...
+
+	link, err := os.Readlink(zoneLink)
 	if err != nil {
 		log.Println("Error finding time zone, ", err)
-		return "", err
+		return "", "", err
 	}
 
-	return link, nil
+	// extract the timezone and the zoneInfoDir from the full path
```
Comment (collinbrake) [inline system/timezone.go]: @cbrake I used two regular expressions here in order to have the flexibility to extract either a file at the top level or a file with 1+ directories in its path.

**Comment by collinbrake** [comment]
@cbrake This is ready to be merged if you wouldn't mind reviewing it.

**Comment by cbrake** [comment]
thanks for the improvements!

**Pull request merged**